lib/addressable/uri.rb

```diff
@@ -2356,5 +2363,10 @@ def split_path(path)
       splitted << EMPTY_STR if path.end_with? SLASH
       splitted
     end
+
+    def reset_uri_string_and_hash
```

Comment:
I’m happy to rename the method and add YARD documentation. I’ll do that now. I was a bit fast and loose with the naming an documentation because it’s a private method.